File: backend/app/api/v1/routers/executors.py
from fastapi import APIRouter, Depends

from app.api.deps import get_current_user, get_executor_service
from app.schemas.executor import ExecutorCreate, ExecutorOut
from app.services.executor import ExecutorService

router = APIRouter(prefix="/executors", tags=["executors"])


@router.get("", response_model=list[ExecutorOut], dependencies=[Depends(get_current_user)])
def list_executors(
    org_unit_id: int,
    service: ExecutorService = Depends(get_executor_service),
):
    """
    List executors for a given org unit (şöbə).
    """
    return service.list_by_org_unit(org_unit_id=org_unit_id)


@router.post("", response_model=ExecutorOut, dependencies=[Depends(get_current_user)])
def create_executor(
    payload: ExecutorCreate,
    service: ExecutorService = Depends(get_executor_service),
):
    """
    Create a new executor under a specific org unit.
    """
    return service.create(full_name=payload.full_name, org_unit_id=payload.org_unit_id)

File: frontend/src/api/executors.ts
import apiClient from './client';

export interface Executor {
  id: number;
  full_name: string;
  org_unit_id: number;
}

export const getExecutors = async (params: {
  org_unit_id: number;
}): Promise<Executor[]> => {
  const response = await apiClient.get('/executors', { params });
  return response.data;
};

export const createExecutor = async (data: {
  full_name: string;
  org_unit_id: number;
}): Promise<Executor> => {
  const response = await apiClient.post('/executors', data);
  return response.data;
};

